packages/@aws-cdk/aws-ivs/README.md

```diff
@@ -9,8 +9,62 @@
 ---
 <!--END STABILITY BANNER-->
 
+Amazon Interactive Video Service (Amazon IVS) is a managed live streaming
+solution that is quick and easy to set up, and ideal for creating interactive
+video experiences. Send your live streams to Amazon IVS using streaming software
+and the service does everything you need to make low-latency live video
+available to any viewer around the world, letting you focus on building
+interactive experiences alongside the live video. You can easily customize and
+enhance the audience experience through the Amazon IVS player SDK and timed
+metadata APIs, allowing you to build a more valuable relationship with your
+viewers on your own websites and applications.
+
 This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.
 
+### Channels
+
+An Amazon IVS channel stores configuration information related to your live
+stream. You first create a channel and then contribute video to it using the
+channel’s stream key to start your live stream.
+
+You can create a channel
+
+```ts
+let myChannel = new ivs.Channel(this, 'Channel');
+```
+
+You can reference an existing channel, for example, if you need to create a
+stream key for an existing channel
+
+```ts
+let myChannel = ivs.Channel.fromChannelArn(this, 'Channel', myChannelArn)
+```
+
+You can create stream keys
+
+```ts
+new ivs.StreamKey(this, 'StreamKey', {
+  channel: myChannel,
+});
+```
```

Comment:
so if the usage described earlier in this README says
> You first create a channel and then contribute video to it using the
channel’s stream key to start your live stream.

1. how does that work? does a user create a stream and a key and then they can use it?
2. should it be an API on `stream`  (i.e. `stream.addStreamKey`)
3. is there a use case where the stream keys need to be created / used independently?

i.e. `stream.addStreamKey(...)`

per the CloudFormation documentation, it feels like that's the heavy lifting the CDK should do for you:
> The Amazon IVS Channel resource does not create a stream key; to create a stream key, use the StreamKey resource instead.

Reply:
Potentially, it's possible that users would rather generate stream keys at the point they want to go live and destroy them after they're done. That's technically the most secure way of doing it, however users also are likely to have a long lived key so I can see it working either way.

Reply:
the `Cfn*` classes that are auto-generated based on the spec already enable these capabilities as a 1:1 mapping between resources and properties i.e: [CfnStreamKey](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ivs.CfnStreamKey.html). As a layer 2 (L2) construct, we should strive to offer intent-based APIs that further simplify the user experience.

Reply:
Right, and I think we can do that once we know what the user experience should be, which is driven by our customers. But getting people using the L2 constructs early means they won't have to change it as and when we add whatever new functionality we decide.

Reply:
That said, I have no problem adding this